src/librepgp/stream-armor.cpp

```diff
@@ -973,48 +973,35 @@ armored_dst_set_line_length(pgp_dest_t *dst, size_t llen)
 bool
 is_armored_source(pgp_source_t *src)
 {
-    uint8_t buf[128];
-    size_t  read = 0;
+    uint8_t buf[strlen(ST_ARMOR_BEGIN)];
```

Comment:
@ni4 you were saying previously that there might be EOL and other whitespace prepending `ST_ARMOR_BEGIN`, this is why `strstr` is used to determine the type of source? And this whitespace is limited by 100-something bytes? Is this a limitation by the protocol?

Reply:
@rrrooommmaaa Yep. This is not a protocol limitation but heuristic, allowing to work in most of the cases. 128 bytes are picked to be enough for most of the cases.